programs/util.c

```diff
@@ -90,6 +90,10 @@ int g_utilDisplayLevel;
 int UTIL_requireUserConfirmation(const char* prompt, const char* abortMsg,
                                           const char* acceptableLetters) {
     int ch, result;
+    /* If input is presented via stdin, dont use prompt as it may swallow characters */
+    if (!IS_CONSOLE(stdin))
```

Comment:
Technically, this test checks if `stdin` is the console.

This is different from the comment, which states that it tests if input is presented via `stdin`.

Example : 
`zstd FILE < /dev/null` : input is _not_ `stdin`, yet `stdin` is not the console either.

I'm fine with the proposed policy : if `stdin` is not the console, then it's not possible to ask for user confirmation via the console, hence `UTIL_requireUserConfirmation()` automatically fails. Just fix the comment.

This policy works as long as we don't support "interactive input" (aka, input defined directly on console),
such as `gzip > dest.gz` : in this scenario, input is `stdin`, yet `stdin` is still the console.

As mentioned, `zstd` doesn't support interactive input, 
so `if (!IS_CONSOLE(stdin))` is fine.
Note though that if we were to revisit that design decision to allow some interactive mode in the future, it would impact this code path.
Might be worth a comment.

Reply:
There is another thread about this above.